src/file/file.rpc.ts

```diff
@@ -41,30 +48,45 @@ export class FileServer {
     // Register the health service
     this.grpcHealthCheck = new GrpcHealthCheck(healthCheckStatusMap);
     this.server.addService(HealthService, this.grpcHealthCheck);
+    this.server.addService(
+      file_proto.FileService.service,
+      {
+        GenerateKey: this.wrapper(this.generateKey),
+        CreateUpload: this.wrapper(this.createUpload),
+        UpdateUploadID: this.wrapper(this.updateUpload),
+        GetUploadByID: this.wrapper(this.getUploadByID),
+        DeleteUploadByID: this.wrapper(this.deleteUploadByID),
+        GetFileByID: this.wrapper(this.getFileByID),
+        GetFileByKey: this.wrapper(this.getFileByKey),
+        GetFilesByFolder: this.wrapper(this.getFilesByFolder),
+        CreateFile: this.wrapper(this.createFile),
+        DeleteFile: this.wrapper(this.deleteFile),
+        IsAllowed: this.wrapper(this.isAllowed),
+      }
+    );
+    this.server.bind(`0.0.0.0:${port}`, grpc.ServerCredentials.createInsecure());
+  }
 
-    this.server.addService(file_proto.FileService.service, {
-      GenerateKey: this.generateKey,
-      CreateUpload: this.createUpload,
-      UpdateUploadID: this.updateUpload,
-      GetUploadByID: this.getUploadByID,
-      DeleteUploadByID: this.deleteUploadByID,
-      GetFileByID: this.getFileByID,
-      GetFileByKey: this.getFileByKey,
-      GetFilesByFolder: this.getFilesByFolder,
-      CreateFile: this.createFile,
-      DeleteFile: this.deleteFile,
-      IsAllowed: this.isAllowed,
-    });
+  private wrapper (func: any) : any {
+    return async (call:any, callback:any) => {
+      try {
+        apm.startTransaction(func.name, 'monitoringFS');
```

Comment:
`func.name` is equal to just the function's name without the service name like `createUpload` where it should be like `/upload.Upload/UploadMedia`, the format is `/${proto_package_name}.${proto_service_name}/${proto_service_function_name}`.

Reply:
so for example: file.file-service.generateKey ?

Reply:
@Shahar-Y No. It should be '/file.FileService/GenerateKey '. There's a reason the protobuf exists...

Reply:
done.

Reply:
@Shahar-Y `func.name` is different than the gRPC name of the method, for example `generateKey` in code implementation but the gRPC method is defined in the protobuf as `GenerateKey`, so it's not done.